checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -216,21 +224,21 @@ def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | s
 
         idx_to_change = new_key or new_value
         self._add_index_to_block_properties(new_resource, idx_to_change)
-        self.local_graph.vertices.append(new_resource)
+        if resource_idx is not None:
+            self.local_graph.vertices[resource_idx] = new_resource
+        else:
+            self.local_graph.vertices.append(new_resource)
 
-    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], main_resource: TerraformBlock) -> None:
+    def _create_new_resources_foreach(self, statement: list[str] | dict[str, Any], block_idx: int) -> None:
+        main_resource = self.local_graph.vertices[block_idx]
         if isinstance(statement, list):
-            for new_value in statement:
-                self._create_new_resource(main_resource, new_value, new_key=new_value)
+            for i, new_value in enumerate(statement):
+                resource_idx = block_idx if i == 0 else None
```

Comment:
```suggestion
                resource_idx = block_idx if i == 0 else -1
```
Let's keep indices with their original type